lametro/models.py

```diff
@@ -166,6 +166,60 @@ def board_report(self):
 
         return br
 
+    @property
+    def actions_and_agendas(self):
+        '''
+        Return list of dictionaries, each representing an action or an agenda on
+        which the given bill appears, in the format:
+
+        {
+            'date': <datetime>,
+            'description': <str>,
+            'event': <LAMetroEvent>,
+            'organization': <Organization>
+        }
+        '''
+        actions = self.actions.all()
+        events = LAMetroEvent.objects.filter(agenda__related_entities__bill=self)
+
+        data = []
+
+        for action in actions:
+            event = LAMetroEvent.objects.get(\
+                participants__entity_type='organization',\
+                participants__name=action.organization,
+                start_time__date=action.date)
+
+            
+            action_dict = {
+                'date': action.date_dt,
+                'description': action.description,
+                'event': event,
+                'organization': action.organization
+            }
+            
+            data.append(action_dict)
+
+        for event in events:
+            # Use a description of "SCHEDULED"
+            org = event.participants.first()
+            event_dict = {
+                'date': event.start_time.date(),
+                'description': 'SCHEDULED',
+                'event': event,
+                'organization': org
+            }
+
+            data.append(event_dict)
+
+        sorted_data = sorted(data, key=lambda x:(x['date'],\
```

Comment:
Can you leave a brief comment about this sort, namely what the desired outcome is and what's going on with the `upper()` and `lower()`?